src/Carnac/KeysController.cs

```
@@ -0,0 +1,106 @@
+                        @----|
+                             @-----o|
+                    ---------------x|
+                    */
+                    return message.Updated
+                        .StartWith(Unit.Default)
+                        .Select(_ => Observable.Timer(fiveseconds, concurrencyService.Default))
+                        .Switch()
+                        .Select(_ => message)
+                        .Take(1);
+                })
+                .Select(m => Tuple.Create(m, ActionType.FadeOut));
+
+            // Finally we just put a one second delay on the messages from the fade out stream and flag to remove.
+            var removeMessageStream = fadeOutMessageStream
+                .Delay(onesecond, concurrencyService.Default)
+                .Select(m => Tuple.Create(m.Item1, ActionType.Remove));
+            
+            var actionStream = removeMessageStream.Merge(fadeOutMessageStream).Merge(addMessageStream);
+
+            actionSubscription = actionStream
+                .ObserveOn(concurrencyService.UiScheduler)
+                .SubscribeOn(concurrencyService.UiScheduler) // Because we mutate message state we need to do everything on UI thread. 
+                                                             // If we introduced a 'Update' action to this feed we could remove mutation from the stream
+                .Subscribe(action =>
+            {
+                switch (action.Item2)
+                {
+                    case ActionType.Add:
+                        keys.Add(action.Item1);
+                        break;
+                    case ActionType.Remove:
+                        keys.Remove(action.Item1);
+                        break;
+                    case ActionType.FadeOut:
+                        action.Item1.IsDeleting = true;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }, ex =>
```

Swallowing the errors feels bad, do we have any sort of logging at the moment to capture this?